packages/driver/src/cy/multi-domain/index.ts

```diff
@@ -84,16 +87,12 @@ export function addCommands (Commands, Cypress: Cypress.Cypress, cy: Cypress.cy,
         const _reject = (err) => {
           cleanup()
           log.error(err)
-          if (typeof err === 'object') {
-            err.onFail = () => {}
-          }
-
           reject(err)
         }
 
         const onQueueFinished = ({ err, subject, unserializableSubjectType }) => {
```

Comment:
What would you think of checking for the special object 'err' in the communicator's event listener and reifying the error there for any events that return the err keyword. If it's a concern that someone may return err, we could have a registry of events that this should be applied to... then you would have just one place to apply the reify logic (i have another event with an error i'd like to have this applied to.

https://github.com/cypress-io/cypress/blob/feature-multidomain/packages/driver/src/multi-domain/communicator.ts#L82

Reply:
I like this idea. Then it mirrors how it's handled on the spec bridge side.

Reply:
I like this idea, but the only thing I think that might be a problem is setting the correct stack. I am not sure how we can get the correct `userInvocationStack` in the `communicator` unless we use `current`. Any ideas?

Reply:
We're already using `current` to the get the `userInvocationStack`. It's guaranteed to be `switchToDomain` because we don't resolve `switchToDomain` until either the secondary queue finishes or there's an error, in which case the queue is stopped for good. 

One option is we could pass the userInvocationStack into the spec bridge with `run:domain:fn` and replace the stack before sending it back, then we don't need to worry about it when reifying the error. This also addresses your question about [this comment](https://github.com/cypress-io/cypress/blob/standardize_error_and_object_serialization/packages/driver/src/multi-domain/events/logs.ts#L16). I think that may be necessary because the logs bypass `switchToDomain` and go straight to the runner. Though now that I think about it, I'm not sure if we want the log to have the `userInvocationStack`, the original stack, or a combination of both. But no need to worry about that for right now. Just pointing out that this solution (passing the userInvocationStack to the spec bridge) may be needed for that use case as well.

Reply:
OK I'm going to give this a shot. I'll pass `userInvocationStack` into `run:domain:fn` to set the invocation stack on the `primarytDomainCommunicator` and look for a special `err` key, and if its truthy, reify it.